### DISCUSSION TITLE
How to make button styles behave like anchor styles

### DISCUSSION THREAD

**Discussion by KaragiannidesAgapios** [tailwindlabs/tailwindcss#13273] [Help] [answered]
I have a problem that i can't find a solution.
I want to somehow manage to copy an anchor's style to a button in smaller screens.
https://play.tailwindcss.com/m3EJQLOvhg
Please make sure you resize in order the text starts wrapping to the next line. The button falls all together to the next line while the anchor tag, wraps the text.

**We want to achieve with button element what is happening with anchor element.**

I thought this will be easier to achieve but i didn't manage...
If anyone has any ideas i'm open.
Thank you.

<img width="464" alt="Screenshot 2024-03-18 at 11 48 56" src="https://github.com/tailwindlabs/tailwindcss/assets/27273385/ff4c4f6d-5832-451b-a554-0f06d450203a">

**Comment by maxlshk** [comment, marked as answer]
It is basically impossible to make button behave like an anchor, because button has hidden inner element, that behaves like a block. We could disable the replaced element-ness if the inner box is not block-level, or some such.

Basically like this, you can try to unstyle a button:

```
button { appearance: none; display: inline; font: inherit; background: initial; border: initial; padding: initial }
button::part(button-contents) { display: inline }
```
But even after this it is unclear how to resolve the issue of button trying to stay together and jumping on the other row, behaving as `inline-block` rather than just `inline`.

There are examples of how to make a button look like a paragraph: https://codepen.io/micahgodbolt/pen/zmrxQd
And docs about specification for the ‘appearance’ CSS property and default rendering of form controls: https://docs.google.com/document/d/1FE5YIoirPKxYbbnMd8kS6w39M8bzLTl5tf4wwOxR1wc/edit#heading=h.xcpgotfgwkdq
if this will be helpful for you.

Also there was a [discussion ](https://github.com/w3c/csswg-drafts/issues/3226)about how to make a button behave like a real inline.

So as a conclusion I think it would be easier for you to use anchors instead of buttons if you want to achieve this functionality

**Reply by KaragiannidesAgapios** [reply]
Well, thank you very much.
I didn't know that it was that complicated.
